pkg/risk/riskcontrol/position.go

```diff
@@ -34,8 +39,30 @@ func NewPositionRiskControl(orderExecutor bbgo.OrderExecutorExtended, hardLimit,
 		sliceQuantity: quantity,
 	}
 
-	control.OnReleasePosition(func(quantity fixedpoint.Value, side types.SideType) {
-		pos := orderExecutor.Position()
+	// register position update handler: check if position is over the hard limit
+	orderExecutor.TradeCollector().OnPositionUpdate(func(position *types.Position) {
+		if fixedpoint.Compare(position.Base, hardLimit) > 0 {
+			log.Infof("position %f is over hardlimit %f, releasing position...", position.Base.Float64(), hardLimit.Float64())
+			control.EmitReleasePosition(position.Base.Sub(hardLimit), types.SideTypeSell)
+		} else if fixedpoint.Compare(position.Base, hardLimit.Neg()) < 0 {
+			log.Infof("position %f is over hardlimit %f, releasing position...", position.Base.Float64(), hardLimit.Float64())
+			control.EmitReleasePosition(position.Base.Neg().Sub(hardLimit), types.SideTypeBuy)
+		}
+	})
+
+	return control
+}
+
+func (p *PositionRiskControl) Initialize(ctx context.Context, session *bbgo.ExchangeSession) {
+	p.activeOrderBook = bbgo.NewActiveOrderBook("")
+	p.activeOrderBook.BindStream(session.UserDataStream)
+
+	p.OnReleasePosition(func(quantity fixedpoint.Value, side types.SideType) {
+		if err := session.Exchange.CancelOrders(ctx, p.activeOrderBook.Orders()...); err != nil {
```

Comment:
you can use orderExecutor to cancel the orders

Reply:
ok